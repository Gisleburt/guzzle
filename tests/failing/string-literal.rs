use guzzle::Guzzle;

#[derive(Default, Guzzle)]
struct GuzzleExample {
    /// These keys should be string literals.
    #[guzzle(keys = [one, two])]
    listed_keys: String,
}

fn main() {
    // These are our keys and values
    let test_data: Vec<(&str, String)> = vec![
        ("one", "1".to_string()),
        ("two", "2".to_string()),
    ];

    // Create our object
    let mut guzzle_example = GuzzleExample::default();

    // Feed our keys and values to our object, capturing any that weren't consumed
    let remaining_data: Vec<(&str, String)> = test_data
        .into_iter()
        .filter_map(|v| guzzle_example.guzzle(v))
        .collect();

    // All appropriate fields are now set
    assert_eq!(guzzle_example.listed_keys, "2".to_string());
}
