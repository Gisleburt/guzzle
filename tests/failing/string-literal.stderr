error: expected literal
 --> $DIR/string-literal.rs:6:22
  |
6 |     #[guzzle(keys = [one, two])]
  |                      ^^^

error[E0599]: no method named `guzzle` found for type `GuzzleExample` in the current scope
  --> $DIR/string-literal.rs:23:40
   |
4  | struct GuzzleExample {
   | -------------------- method `guzzle` not found for this
...
23 |         .filter_map(|v| guzzle_example.guzzle(v))
   |                                        ^^^^^^ method not found in `GuzzleExample`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `guzzle`, perhaps you need to implement it:
           candidate #1: `guzzle::Guzzle`
